### PR TITLE
Unit-tests for collector

### PR DESCRIPTION
**Pull request by v-nikulichev**
This is initial commit setting up CMake and Google Test. It also includes a simple test for `ConfigParser`.
Commit summary:
- **Remove old tests**: Collector was completely reworked since the old tests were updated last time. We have nothing to do with them.
- **Move default values to constants**: Initialize `Config` members from constexpr values.
- **Static library collector_common**: Move core modules to static library. Tests and the application will be linked with it, no need to recompile .cpp files twice or more.
- **Static loggers and config for tests**: We don't use logger and config objects residing in `WorkerApplication.cpp`. This way we're able to change values in runtime (and don't save any logs in `/var/log`).
- **CMake settings for tests**: Main rules how to build the tests. This also adds Google Test submodule.
- **Test for ConfigParser**: The first test working in the new environment.
- **Enable testing**: Uncomment test instructions in root `CMakeLists.txt`.

### CODE REVIEW

**Comment by shindo** [comment]
Ты не обновил `debian/rules`, из-за чего тесты не запускаются после сборки.
Туда нужно добавить определение `DEB_MAKE_CHECK_TARGET`.

Учти, что `debian/rules` в мастере/девелопе изменился, также там добавились питонячие тесты.
Поэтому вам стоит определиться с тем, как вы разрабатываетесь вместе (связь твоей ветки `ccp-worker-gcd` и `master`/`develop`).

File: tests/collector/util/app.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+std::unique_ptr<ioremap::elliptics::logger_base> s_logger;
+std::unique_ptr<ioremap::elliptics::logger_base> s_elliptics_logger;
+
+Config s_config;
+
+void open_logger()
+{
+    s_logger.reset(new ioremap::elliptics::file_logger(
+        "/dev/stdout", ioremap::elliptics::log_level(0)));
```
Comment (shaitan) [inline tests/collector/util/app.cpp]: Please, use some of `DNET_LOG_*` here.

Comment (v-nikulichev) [inline tests/collector/util/app.cpp]: https://github.com/v-nikulichev/mastermind/commit/6ed1f8fb618ef64a523e823c848fe057b2b83d5d

File: tests/collector/config_parser/config_parser.cpp [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    }
+};
+
+TestConfig s_config;
+
+// Generate JSON config.
+std::string generate_json()
+{
+    rapidjson::StringBuffer buf;
+    rapidjson::Writer<rapidjson::StringBuffer> writer(buf);
```
Comment (shaitan) [inline tests/collector/config_parser/config_parser.cpp]: I supposed that we agreed that you will use raw string literals with final json string, weren't we?

File: tests/collector/config_parser/config_parser.cpp [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+    Config config;
+    config.~Config();
+    // Write junk bytes and check members initialization afterwards.
+    memset(&config, 0x5A, sizeof(config));
+    new (&config) Config;
+
+    // Workaround for undefined references in macros
+    // EXPECT_EQ(Config::Default::xxx, config.xxx)
+    auto default_monitor_port = Config::Default::monitor_port;
```
Comment (shaitan) [inline tests/collector/config_parser/config_parser.cpp]: This workaround looks at least strange, can `undefined references` issue be fixed in another way?

File: tests/collector/config_parser/config_parser.cpp (hunk 2 above)
Comment (v-nikulichev) [inline tests/collector/config_parser/config_parser.cpp]: https://github.com/v-nikulichev/mastermind/commit/daf95c083d6c8d2b3f9e3ed12c490d5d304cf704

File: tests/collector/config_parser/config_parser.cpp (hunk 3 above)
Comment (v-nikulichev) [inline tests/collector/config_parser/config_parser.cpp]: https://github.com/v-nikulichev/mastermind/commit/9c4004b036008a28f46f4ae94b5c71706ded8207

**Comment by shindo** [comment]
lgtm

**Pull request merged**